### PR TITLE
r/glue_partition - add nil check for `schema_reference`

### LINKED ISSUE #17679: Upgrade from 3.22 to 3.28 results in an error
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)
Not clear

### Expected Behavior

Provider version upgrade should not result in an error

### Actual Behavior

Upgrading provider from version 3.22 to version 3.28 results in an error

```
Error: error setting storage_descriptor: Invalid address to set: []string{"storage_descriptor", "0", "schema_reference"}
```

### Steps to Reproduce
Not clear, but rolling provider back to 3.22 solves the problem

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #17679

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSGluePartition_'
--- PASS: TestAccAWSGluePartition_disappears (42.79s)
--- PASS: TestAccAWSGluePartition_basic (46.65s)
--- PASS: TestAccAWSGluePartition_multipleValues (49.25s)
--- PASS: TestAccAWSGluePartition_parameters (117.59s)
--- PASS: TestAccAWSGluePartition_disappears_table (41.50s)
```

```
$ make testacc TESTARGS='-run=TestAccAWSGlueCatalogTable'
--- PASS: TestAccAWSGlueCatalogTable_disappears_database (32.62s)
--- PASS: TestAccAWSGlueCatalogTable_disappears (39.25s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_SkewedInfo_EmptyConfigurationBlock (42.82s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_EmptyConfigurationBlock (43.48s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_SerDeInfo_EmptyConfigurationBlock (43.54s)
--- PASS: TestAccAWSGlueCatalogTable_columnParameters (43.88s)
--- PASS: TestAccAWSGlueCatalogTable_partitionIndexesMultiple (46.30s)
--- PASS: TestAccAWSGlueCatalogTable_full (47.73s)
--- PASS: TestAccAWSGlueCatalogTable_partitionIndexesSingle (48.69s)
--- PASS: TestAccAWSGlueCatalogTable_basic (50.23s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_schemaReference (62.31s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_schemaReferenceArn (63.68s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_SerDeInfo_UpdateValues (77.36s)
--- PASS: TestAccAWSGlueCatalogTable_update_addValues (77.54s)
--- PASS: TestAccAWSGlueCatalogTable_update_replaceValues (79.08s)
```

### CODE REVIEW

File: .changelog/17702.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+```release-note:bug
+resource/aws_glue_partition: Add nil check for `schema_reference`
```
Comment (ewbankkit) [inline .changelog/17702.txt:2]:
```suggestion
resource/aws_glue_catalog_table: Add nil check for `schema_reference`
```

**Review by ewbankkit — CHANGES_REQUESTED** [review]
Other than the release note change, LGTM.

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSGluePartition_\|TestAccAWSGlueCatalogTable_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSGluePartition_\|TestAccAWSGlueCatalogTable_ -timeout 120m
=== RUN   TestAccAWSGlueCatalogTable_basic
=== PAUSE TestAccAWSGlueCatalogTable_basic
=== RUN   TestAccAWSGlueCatalogTable_columnParameters
=== PAUSE TestAccAWSGlueCatalogTable_columnParameters
=== RUN   TestAccAWSGlueCatalogTable_full
=== PAUSE TestAccAWSGlueCatalogTable_full
=== RUN   TestAccAWSGlueCatalogTable_update_addValues
=== PAUSE TestAccAWSGlueCatalogTable_update_addValues
=== RUN   TestAccAWSGlueCatalogTable_update_replaceValues
=== PAUSE TestAccAWSGlueCatalogTable_update_replaceValues
=== RUN   TestAccAWSGlueCatalogTable_StorageDescriptor_EmptyConfigurationBlock
=== PAUSE TestAccAWSGlueCatalogTable_StorageDescriptor_EmptyConfigurationBlock
=== RUN   TestAccAWSGlueCatalogTable_StorageDescriptor_SerDeInfo_EmptyConfigurationBlock
=== PAUSE TestAccAWSGlueCatalogTable_StorageDescriptor_SerDeInfo_EmptyConfigurationBlock
=== RUN   TestAccAWSGlueCatalogTable_StorageDescriptor_SerDeInfo_UpdateValues
=== PAUSE TestAccAWSGlueCatalogTable_StorageDescriptor_SerDeInfo_UpdateValues
=== RUN   TestAccAWSGlueCatalogTable_StorageDescriptor_SkewedInfo_EmptyConfigurationBlock
=== PAUSE TestAccAWSGlueCatalogTable_StorageDescriptor_SkewedInfo_EmptyConfigurationBlock
=== RUN   TestAccAWSGlueCatalogTable_StorageDescriptor_schemaReference
=== PAUSE TestAccAWSGlueCatalogTable_StorageDescriptor_schemaReference
=== RUN   TestAccAWSGlueCatalogTable_StorageDescriptor_schemaReferenceArn
=== PAUSE TestAccAWSGlueCatalogTable_StorageDescriptor_schemaReferenceArn
=== RUN   TestAccAWSGlueCatalogTable_partitionIndexesSingle
=== PAUSE TestAccAWSGlueCatalogTable_partitionIndexesSingle
=== RUN   TestAccAWSGlueCatalogTable_partitionIndexesMultiple
=== PAUSE TestAccAWSGlueCatalogTable_partitionIndexesMultiple
=== RUN   TestAccAWSGlueCatalogTable_disappears_database
=== PAUSE TestAccAWSGlueCatalogTable_disappears_database
=== RUN   TestAccAWSGlueCatalogTable_disappears
=== PAUSE TestAccAWSGlueCatalogTable_disappears
=== RUN   TestAccAWSGluePartition_basic
=== PAUSE TestAccAWSGluePartition_basic
=== RUN   TestAccAWSGluePartition_multipleValues
=== PAUSE TestAccAWSGluePartition_multipleValues
=== RUN   TestAccAWSGluePartition_parameters
=== PAUSE TestAccAWSGluePartition_parameters
=== RUN   TestAccAWSGluePartition_disappears
=== PAUSE TestAccAWSGluePartition_disappears
=== RUN   TestAccAWSGluePartition_disappears_table
=== PAUSE TestAccAWSGluePartition_disappears_table
=== CONT  TestAccAWSGlueCatalogTable_basic
=== CONT  TestAccAWSGlueCatalogTable_partitionIndexesSingle
=== CONT  TestAccAWSGluePartition_disappears_table
=== CONT  TestAccAWSGluePartition_disappears
=== CONT  TestAccAWSGluePartition_parameters
=== CONT  TestAccAWSGluePartition_multipleValues
=== CONT  TestAccAWSGluePartition_basic
=== CONT  TestAccAWSGlueCatalogTable_disappears
=== CONT  TestAccAWSGlueCatalogTable_disappears_database
=== CONT  TestAccAWSGlueCatalogTable_partitionIndexesMultiple
=== CONT  TestAccAWSGlueCatalogTable_StorageDescriptor_SerDeInfo_EmptyConfigurationBlock
=== CONT  TestAccAWSGlueCatalogTable_StorageDescriptor_schemaReferenceArn
=== CONT  TestAccAWSGlueCatalogTable_StorageDescriptor_schemaReference
=== CONT  TestAccAWSGlueCatalogTable_StorageDescriptor_SkewedInfo_EmptyConfigurationBlock
=== CONT  TestAccAWSGlueCatalogTable_StorageDescriptor_SerDeInfo_UpdateValues
=== CONT  TestAccAWSGlueCatalogTable_update_addValues
=== CONT  TestAccAWSGlueCatalogTable_StorageDescriptor_EmptyConfigurationBlock
=== CONT  TestAccAWSGlueCatalogTable_update_replaceValues
=== CONT  TestAccAWSGlueCatalogTable_full
=== CONT  TestAccAWSGlueCatalogTable_columnParameters
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_SerDeInfo_EmptyConfigurationBlock (53.59s)
--- PASS: TestAccAWSGlueCatalogTable_disappears_database (53.92s)
--- PASS: TestAccAWSGluePartition_disappears_table (63.16s)
--- PASS: TestAccAWSGlueCatalogTable_disappears (63.16s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_SkewedInfo_EmptyConfigurationBlock (63.71s)
--- PASS: TestAccAWSGluePartition_disappears (63.86s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_EmptyConfigurationBlock (63.91s)
--- PASS: TestAccAWSGlueCatalogTable_partitionIndexesSingle (68.78s)
--- PASS: TestAccAWSGlueCatalogTable_partitionIndexesMultiple (69.89s)
--- PASS: TestAccAWSGlueCatalogTable_columnParameters (69.89s)
--- PASS: TestAccAWSGlueCatalogTable_basic (70.37s)
--- PASS: TestAccAWSGlueCatalogTable_full (70.44s)
--- PASS: TestAccAWSGluePartition_multipleValues (70.54s)
--- PASS: TestAccAWSGluePartition_basic (71.03s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_schemaReferenceArn (74.68s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_schemaReference (76.16s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_SerDeInfo_UpdateValues (81.00s)
--- PASS: TestAccAWSGlueCatalogTable_update_replaceValues (82.49s)
--- PASS: TestAccAWSGlueCatalogTable_update_addValues (84.87s)
--- PASS: TestAccAWSGluePartition_parameters (98.01s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	98.135s
```

**Review by ewbankkit — APPROVED** [review]

File: .changelog/17702.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+```release-note:bug
+resource/aws_glue_catalog_table: Add nil check for `schema_reference`
```
Comment (bflad) [inline .changelog/17702.txt:2]: To make this note more practitioner friendly, recommend saying what the change is doing for them:

```suggestion
resource/aws_glue_catalog_table: Prevent `schema_reference` attribute error
```

**Review by bflad — APPROVED** [review]
Looks great, thank you, @DrFaust92 🚀 

Output from acceptance testing in AWS Commercial:

```
--- PASS: TestAccAWSGlueCatalogTable_basic (31.10s)
--- PASS: TestAccAWSGlueCatalogTable_columnParameters (33.94s)
--- PASS: TestAccAWSGlueCatalogTable_disappears (30.13s)
--- PASS: TestAccAWSGlueCatalogTable_disappears_database (24.49s)
--- PASS: TestAccAWSGlueCatalogTable_full (35.08s)
--- PASS: TestAccAWSGlueCatalogTable_partitionIndexesMultiple (35.00s)
--- PASS: TestAccAWSGlueCatalogTable_partitionIndexesSingle (34.57s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_EmptyConfigurationBlock (31.82s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_schemaReference (38.78s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_schemaReferenceArn (39.47s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_SerDeInfo_EmptyConfigurationBlock (31.91s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_SerDeInfo_UpdateValues (43.66s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_SkewedInfo_EmptyConfigurationBlock (31.99s)
--- PASS: TestAccAWSGlueCatalogTable_update_addValues (45.27s)
--- PASS: TestAccAWSGlueCatalogTable_update_replaceValues (46.23s)

--- PASS: TestAccAWSGluePartition_basic (35.44s)
--- PASS: TestAccAWSGluePartition_disappears (31.50s)
--- PASS: TestAccAWSGluePartition_disappears_table (30.91s)
--- PASS: TestAccAWSGluePartition_multipleValues (35.76s)
--- PASS: TestAccAWSGluePartition_parameters (56.26s)
```

Output from acceptance testing in AWS GovCloud (US):

```
--- FAIL: TestAccAWSGlueCatalogTable_StorageDescriptor_schemaReference (15.54s) # https://github.com/hashicorp/terraform-provider-aws/issues/17449
--- FAIL: TestAccAWSGlueCatalogTable_StorageDescriptor_schemaReferenceArn (14.85s) # https://github.com/hashicorp/terraform-provider-aws/issues/17449
--- PASS: TestAccAWSGlueCatalogTable_basic (32.74s)
--- PASS: TestAccAWSGlueCatalogTable_columnParameters (34.11s)
--- PASS: TestAccAWSGlueCatalogTable_disappears (28.84s)
--- PASS: TestAccAWSGlueCatalogTable_disappears_database (24.42s)
--- PASS: TestAccAWSGlueCatalogTable_full (31.56s)
--- PASS: TestAccAWSGlueCatalogTable_partitionIndexesMultiple (32.82s)
--- PASS: TestAccAWSGlueCatalogTable_partitionIndexesSingle (32.55s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_EmptyConfigurationBlock (30.14s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_SerDeInfo_EmptyConfigurationBlock (30.73s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_SerDeInfo_UpdateValues (47.27s)
--- PASS: TestAccAWSGlueCatalogTable_StorageDescriptor_SkewedInfo_EmptyConfigurationBlock (30.30s)
--- PASS: TestAccAWSGlueCatalogTable_update_addValues (48.12s)
--- PASS: TestAccAWSGlueCatalogTable_update_replaceValues (47.88s)

--- PASS: TestAccAWSGluePartition_basic (34.71s)
--- PASS: TestAccAWSGluePartition_disappears (31.71s)
--- PASS: TestAccAWSGluePartition_disappears_table (30.66s)
--- PASS: TestAccAWSGluePartition_multipleValues (35.11s)
--- PASS: TestAccAWSGluePartition_parameters (64.63s)
```

**Pull request merged**